### PR TITLE
test: move LibvirtDomainIsPersistent from utils

### PR DESCRIPTION
**Pull request by codingben**
Move LibvirtDomainIsPersistent from utils to
migration_test to make utils shorter.

Signed-off-by: Ben Oukhanov <boukhanov@redhat.com>

**Special notes for your reviewer**:

We had [here](https://github.com/kubevirt/kubevirt/pull/7810/files#diff-847cd06c1bf6ef597ea96894f8b32ecb4d074ad1bc785b39871e11fb6ab71b3dL2201):
```
	vmiPod, err := getRunningPodByVirtualMachineInstance(vmi, util2.NamespaceTestDefault)
	if err != nil {
		return false, err
	}
```
Now we're using `GetRunningPodByVirtualMachineInstance` and have `util2.PanicOnError(err)` [there](https://github.com/benukhanov/kubevirt/blob/19e78a2ec267db569acf8c1e662a52332df8d6a1/tests/utils.go#L845).

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: tests/migration_test.go [hunk 1]
```diff
@@ -4128,3 +4128,29 @@ func disableNodeLabeller(node *k8sv1.Node, virtClient kubecli.KubevirtClient) *k
 
 	return node
 }
+
+func LibvirtDomainIsPersistent(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) (bool, error) {
+	vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, util.NamespaceTestDefault)
```
Comment (benukhanov) [inline tests/migration_test.go:4133]: Just a note, we had here:
```
	vmiPod, err := getRunningPodByVirtualMachineInstance(vmi, util2.NamespaceTestDefault)
	if err != nil {
		return false, err
	}
```
Now we have `util2.PanicOnError(err)` [there](https://github.com/benukhanov/kubevirt/blob/19e78a2ec267db569acf8c1e662a52332df8d6a1/tests/utils.go#L845).

File: tests/migration_test.go [hunk 2]
```diff
@@ -4128,3 +4128,29 @@ func disableNodeLabeller(node *k8sv1.Node, virtClient kubecli.KubevirtClient) *k
 
 	return node
 }
+
+func LibvirtDomainIsPersistent(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) (bool, error) {
```
[the remark this answers is not in the document]
Comment (boukhanov) [reply tests/migration_test.go:4132]: Yep, sure.

[the remark this answers is not in the document]
Comment (boukhanov) [reply tests/migration_test.go]: Done.

File: tests/migration_test.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ import (
 	"kubevirt.io/kubevirt/tests/clientcmd"
 	"kubevirt.io/kubevirt/tests/framework/checks"
 	"kubevirt.io/kubevirt/tests/libnode"
-	"kubevirt.io/kubevirt/tests/util"
+	util "kubevirt.io/kubevirt/tests/util"
```
Comment (dankenigsberg) [inline tests/migration_test.go:45]: Is this change intentional/helpful?

**Review by dankenigsberg — COMMENTED** [review]
/ok-to-test

File: tests/migration_test.go (hunk 3 above)
Comment (codingben) [reply tests/migration_test.go:45]: Yes, it's used here:
```
vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, util.NamespaceTestDefault)
```
Go (something that analyzes the code) added that line automatically as `util2 "kubevirt.io/kubevirt/tests/util"` that was used here:
```
vmiPod, err := getRunningPodByVirtualMachineInstance(vmi, util2.NamespaceTestDefault)
```
so I just called it `util`. I see that it can work without `util` and it's not needed (it was used by other code). I'd like to run `make functest` again.

For now please don't let CI run to save resources. I'd like to re-run it locally again on Sunday (couldn't run now in dockerized env because it doesn't work for some reason).

Comment (codingben) [reply tests/migration_test.go]: Removed `util` from there. Ran tests locally successfully. Please re-review it :)

File: tests/migration_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dankenigsberg) [reply tests/migration_test.go:4132]: Did you intentionally make it public again?

P.s I have no idea why the test cares whether the vm is persistent or not. But we can discuss this in another PR.

**Comment by dankenigsberg** [comment]
/lgtm

Though I disagree with the premise of https://github.com/kubevirt/kubevirt/pull/5010 : whether the domain is persistent or transient is an implementation detail, which should not be tested by a functional test.

File: tests/migration_test.go (hunk 1 above)
Comment (enp0s3) [reply tests/migration_test.go:4133]: @codingben Hi. Thank you for your PR! you can add such notes in the PR description under the **special notes for the reviewer**.

**Review by enp0s3 — COMMENTED** [review]
/approve

I've noticed that although its a code clean PR, it is does a small change in the test behavior, since now instead of fail on assertion, the call to `LibvirtDomainIsPersistent` will panic upon failure. However I don't see it as a reason for another review cycle, we can do the refactoring in a follow-up PR. Thank you @codingben!

File: tests/migration_test.go (hunk 1 above)
Comment (codingben) [reply tests/migration_test.go:4133]: Yep, sounds good. Thanks!

**Review by enp0s3 — APPROVED** [review]
